packages/react-positioning/src/usePopper.ts

```diff
@@ -298,86 +110,103 @@ export function usePopper(
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   arrowRef: React.MutableRefObject<any>;
 } {
-  const { enabled = true } = options;
   const isFirstMount = useFirstMount();
+  const { targetDocument } = useFluent();
+  const { enabled = true } = options;
+  const resolvePopperOptions = usePopperOptions(options);
 
-  const popperOriginalPositionRef = React.useRef<string>('absolute');
-  const resolvePopperOptions = usePopperOptions(options, popperOriginalPositionRef);
+  const forceUpdate = useEventCallback(() => {
+    const target = overrideTargetRef.current ?? targetRef.current;
+    if (!canUseDOM || !enabled || !target || !containerRef.current) {
+      return;
+    }
 
-  const popperInstanceRef = React.useRef<PopperInstance | null>(null);
+    const { placement, middleware, strategy } = resolvePopperOptions(target, containerRef.current, arrowRef.current);
 
-  const handlePopperUpdate = useEventCallback(() => {
-    popperInstanceRef.current?.destroy();
-    popperInstanceRef.current = null;
+    Object.assign(containerRef.current.style, { position: strategy });
```

Comment:
Is it correct that we apply styles there? Should we apply them in `.then()`?

Reply:
this is part of the positionFixed hack :)

Reply:
Let's add a comment then 😉